recipes/gtest/all/conanfile.py

```diff
@@ -53,18 +54,31 @@ def _minimum_compilers_version(self):
     def _is_clang_cl(self):
         return self.settings.os == "Windows" and self.settings.compiler == "clang"
 
+    def export_sources(self):
+        if Version(self.version) < "1.12.0":
+            for patch in self.conan_data.get("patches", {}).get(self.version, []):
+                copy(self, patch["patch_file"], self.recipe_folder, self.export_sources_folder)
+
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if Version(self.version) < "1.12.0" and self.settings.build_type != "Debug":
+            del self.options.debug_postfix
 
     def configure(self):
         if self.options.shared:
             try:
                 del self.options.fPIC
             except Exception:
                 pass
-        if self.options.debug_postfix != "deprecated":
-            self.output.warn("gtest/*:debug_postfix is deprecated.")
+        if Version(self.version) >= "1.12.0":
+            if self.options.debug_postfix != "deprecated":
+                self.output.warn("gtest/*:debug_postfix is deprecated in 1.12.0.")
+
+        if Version(self.version) < "1.12.0" and self.settings.build_type == "Debug":
+            if self.options.debug_postfix == "deprecated":
+                # in 1.10.0,  use 'd' as default
+                self.options.debug_postfix = "d"
```

Comment:
```suggestion
```
You can't mask the the option value, because it will generate a very same package ID for both values. Instead, change its default value on `config_options()`.